vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/GlobalExceptionHandler.java

```
@@ -0,0 +1,64 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.vcfdump.server;
+
+import htsjdk.samtools.util.RuntimeIOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.async.AsyncRequestTimeoutException;
+
+import uk.ac.ebi.eva.vcfdump.VariantExporter;
+
+@ControllerAdvice(assignableTypes = VcfDumperWSServer.class)
+public class GlobalExceptionHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(VariantExporter.class);
+
+    @ExceptionHandler({RuntimeException.class, RuntimeIOException.class})
+    public final ResponseEntity handleException(Exception e) {
+        HttpHeaders headers = new HttpHeaders();
+
+        if (e instanceof AsyncRequestTimeoutException) {
+            HttpStatus status = HttpStatus.REQUEST_TIMEOUT;
+            AsyncRequestTimeoutException asyncRequestTimeoutException = (AsyncRequestTimeoutException) e;
+            return handleAsyncRequestTimeoutException(asyncRequestTimeoutException, headers, status);
+        } else if (e instanceof RuntimeException) {
+            HttpStatus status = HttpStatus.SERVICE_UNAVAILABLE;
+            RuntimeException runtimeException = (RuntimeException) e;
+            return handleRuntimeException(runtimeException, headers, status);
+        } else {
+            HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
+            return new ResponseEntity<>(null, headers, status);
```

Do not hide the exception, it will only makes the response more confusing and complicate debugging. If the branch above this one is converted to 500 status, they can probably be merged into a `handleException` method.